packages/app-builder/src/routes/_builder+/d3.tsx

```diff
@@ -0,0 +1,251 @@
+                fill="currentColor"
+                className={outcomeColor[outcome]}
+              />
+            </svg>
+            <span className="text-s text-grey-100 font-medium">
+              {t(`decisions:outcome.${outcome}`)}
+            </span>
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+}
+
+const orderredOutcomes: Outcome[] = [
+  'decline',
+  'block_and_review',
+  'review',
+  'approve',
+];
+
+const outcomeColor = {
+  approve: 'text-green-100',
+  decline: 'text-red-100',
+  review: 'text-yellow-100',
+  block_and_review: 'text-orange-100',
+  unknown: 'text-grey-100',
+};
+
+function ChartInner({
+  data,
+  width,
+  height,
+}: {
+  data: Data['nodes'];
+  width: number;
+  height: number;
+}) {
+  const margin = {
+    top: 37,
+    right: 0,
+    bottom: 0,
+    left: 0,
+  };
+  const gap = 4;
+
+  const foo = d3.index(
+    data,
+    (d) => d.version,
+    (d) => d.outcome,
+  );
+
+  // Determine the series that need to be stacked.
+  const series = d3
+    .stack<[string, d3.InternMap<Outcome, Data['nodes'][number]>], Outcome>()
+    .offset(d3.stackOffsetExpand)
+    .keys(orderredOutcomes)
+    .order(d3.stackOrderNone)
+    .value(([, D], key) => {
+      return D.get(key)?.count ?? 0;
+    })(foo);
+
+  // Prepare the scales for positional and color encodings.
+  const xScale = d3
+    .scaleBand()
+    .domain(d3.union(data.map((d) => d.version)))
+    .range([margin.left, width - margin.right])
+    .paddingInner(0.1);
+
+  const versionOptions = new Map(
+    foo.entries().map(([version, value]) => {
+      const outcomes = d3.sort(
+        value.keys(),
+        (outcome) => -orderredOutcomes.indexOf(outcome),
+      );
+      const outcomeOffset = new Map(
+        outcomes.map((outcome, i) => [outcome, i * gap]),
+      );
+      const numberOfOutcomes = Object.keys(outcomes).length;
+
+      return [
+        version,
+        {
+          outcomeOffset,
+          yScale: d3
+            .scaleLinear()
+            .domain([0, 1])
+            .range([
+              height - (numberOfOutcomes - 1) * gap - margin.bottom,
+              margin.top,
+            ]),
+        },
+      ];
+    }),
+  );
+
+  return (
+    <svg className="border" viewBox={`0 0 ${width} ${height}`}>
+      {/* Vi Labels */}
+      {xScale.domain().map((xi, i) => (
+        <g
+          key={xi}
+          transform={`translate(${xScale.step() * i + xScale.bandwidth() / 2},5)`}
+        >
+          <text
+            className="text-grey-100 text-s font-medium capitalize"
+            textAnchor="middle"
+            alignmentBaseline="text-before-edge"
+            fill="currentColor"
+          >
+            {xi}
+          </text>
+        </g>
+      ))}
+
+      <use
+        href={`${iconsSVGSpriteHref}#arrow-right`}
+        height={40}
+        width={40}
+        x={(width + margin.left - margin.right - 40) / 2}
+        y={(height + margin.top - margin.bottom - 40) / 2}
+      />
```

Comment:
Reuse of an existing svg icon (it's the abstracted SVG code hidden inside `Icon` component)